File: quotebot-go/internal/tech_blogs.go
package internal

import "time"

/*
This simulates a very slow database connection

You are not supposed to touch this file
*/

// ListAllBlogs returns all tech blogs (very slow)
func ListAllBlogs() []string {
	time.Sleep(5 * time.Second)

	return []string{
		"HackerNews",
		"Reddit",
		"TechCrunch",
		"BuzzFeed",
		"TheHuffPost",
		"TMZ",
		"GigaOM",
	}
}

File: quotebot-elixir/lib/tech_blogs.ex
defmodule Quotebot.TechBlogs do
  @moduledoc """
  This simulates a very slow database connection.

  You are not supposed to touch this file.
  """

  def list_all_blogs do
    # Very slow database access
    Process.sleep(5000)

    [
      "HackerNews",
      "Reddit",
      "TechCrunch",
      "BuzzFeed",
      "TheHuffPost",
      "TMZ",
      "GigaOM"
    ]
  end
end
